enterprise/server/remote_execution/containers/docker/docker.go

```diff
@@ -462,7 +472,19 @@ func (r *dockerCommandContainer) exec(ctx context.Context, command *repb.Command
 	}
 
 	defer attachResp.Close() // note: Close() doesn't return an error.
+	go func() {
+		// If the context times out before we return from this func, close the
+		// attachResp -- otherwise copyOutputs() hangs.
+		<-ctx.Done()
+		attachResp.Close()
```

Comment:
is it safe to call attachResp when the func returns before ctx.Done hits?

Reply:
I didn't notice any problems in practice, but wrapped with a `sync.Once` just to be safe

Reply:
resolved offline - attachResp.Close() ultimately calls this code https://cs.opensource.google/go/go/+/refs/tags/go1.18:src/net/http/httputil/persist.go;l=286-292;drc=cbf90b0028265a6195257e31a03aae42d402dd64 which is safe to call more than once